File: 研究方向/算法/LeetCode/CodeStash/35.搜索插入位置.cpp
/*
 * @lc app=leetcode.cn id=35 lang=cpp
 *
 * [35] 搜索插入位置
 */

// @lc code=start
class Solution {
public:
  int searchInsert(vector<int> &nums, int target) {
    int pivot, left = 0, right = nums.size() - 1;
    while (left <= right) {
      pivot = left + ((right - left) >> 1);
      if (nums[pivot] < target)
        left = pivot + 1;
      else if (nums[pivot] > target)
        right = pivot - 1;
      else
        return pivot;
    }
    return left;
  }
};
// @lc code=end

File: 学习路线/算法/LeetCode/CodeStash/35.搜索插入位置.py
#
# @lc app=leetcode.cn id=35 lang=python3
#
# [35] 搜索插入位置
#

# @lc code=start
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums) - 1
        while left <= right:
            p = left + ((right - left) >> 1)
            if nums[p] < target:
                left = p + 1
            else:
                right = p - 1
        return left
# @lc code=end

